consumer/apis/dropbox/oauth_20_dropbox_api.e

```diff
@@ -0,0 +1,66 @@
+note
+	description: "Summary description for {OAUTH_20_DROPBOX_API}."
+	date: "$Date$"
+	revision: "$Revision$"
+	EIS: "name=Dropbox OAuth 2.0", "src=https://www.dropbox.com/developers/blog/45/using-oauth-20-with-the-core-api", "protocol=url"
+class
+	OAUTH_20_DROPBOX_API
+
+inherit
+
+	OAUTH_20_API
+		redefine
+			access_token_extractor,
+			access_token_verb
+		end
+
+feature -- Access
+
+	access_token_extractor: ACCESS_TOKEN_EXTRACTOR
+		do
+			create {JSON_TOKEN_EXTRACTOR} Result
+		end
+
+	access_token_verb: READABLE_STRING_GENERAL
+		do
+			Result := "POST"
+		end
+
+	access_token_endpoint: READABLE_STRING_GENERAL
+			-- Url that receives the access token request
+		do
+			create {STRING_32} Result.make_from_string ("https://api.dropbox.com/1/oauth2/token")
+		end
+
+	authorization_url (config: OAUTH_CONFIG): detachable READABLE_STRING_GENERAL
+			-- Url where you should redirect your users to authneticate
+		local
+			l_result: STRING_32
+		do
+			if attached config.state as l_state then
+				create {STRING_32} l_result.make_from_string (Template_authorize_url)
```

Comment:
Result is not assigned with l_result ..
but I would recommend to return detachable STRING_8   (url should not have unicode)
and use Result , instead of l_result